packages/angular_devkit/build_angular/src/differential-loading/is-differential-loading-needed.ts

```diff
@@ -0,0 +1,138 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import * as browserslist from 'browserslist';
+import * as path from 'path';
+import { ScriptTarget } from 'typescript';
+
+const es6ModuleSupport = require('./es6-module-support.json');
+const supportTable = convertSupportTableToBrowserVersionTuples(es6ModuleSupport);
+
+export function isDifferentialLoadingNeeded(
+  tsConfigFileName: string, query: string[] | undefined = undefined) {
+  const es5SupportNeeded = isEs5BrowserSupportNeeded(tsConfigFileName, query);
+  const target = getTargetVersion(tsConfigFileName);
+
+  return es5SupportNeeded && target !== ScriptTarget.ES5;
+}
+
+export function getTargetVersion(tsConfigFileName: string): ScriptTarget {
+  return _getTargetVersion(tsConfigFileName, new Set<string>());
+}
+
+function _getTargetVersion(tsConfigFileName: string, visited: Set<string>): ScriptTarget {
```

Comment:
We shouldn't be re-creating the ts parsing parsing logic of typescript. You can simply use `readTsconfig` method.